### ISSUE TITLE
Collision and visual do not align for d435 

### ISSUE THREAD

**Issue by samibouziri** [realsenseai/realsense-ros#2969]
Hello,

I noticed that the mesh and collision box do not align for the the d435 as depicted in the image bellow.
![image](https://github.com/IntelRealSense/realsense-ros/assets/79418773/0dfeafe5-f7c7-41c2-a848-899fd1cd9857)
The origin of that problem is that the x coordinate of the origin of the collision box for link ${name}_link seems to be wrong. It should be -d435_cam_depth/2+d435_zero_depth_to_glass + d435_glass_to_front instead of 0.
Same goes for x coordinate of the visual if we are not using the mesh.

**Comment by MartyG-RealSense** [comment]
Hi @samibouziri  Are you using a URDF, please?  If you are then the above image looks unusual for a D435.  A D435 URDF should look like the one at https://github.com/IntelRealSense/realsense-ros/issues/459#issuecomment-420505591

![image](https://github.com/IntelRealSense/realsense-ros/assets/41145062/ddede1ba-49e6-4d58-8726-8eaf13f7ecb3)

**Comment by samibouziri** [comment]
Yes I am using this [urdf](https://github.com/IntelRealSense/realsense-ros/blob/ros2-development/realsense2_description/urdf/_d435.urdf.xacro) I am just enabling the visualization of the mesh and the corresponding collision box in the same picture to demonstrate the offset.
here is the mesh alone:
![image](https://github.com/IntelRealSense/realsense-ros/assets/79418773/053ce108-959c-452a-ae74-809d4ffd06df)
here is the collision box alone:
![image](https://github.com/IntelRealSense/realsense-ros/assets/79418773/4f178c9b-2dea-429a-b0b7-1c46b9369ad5)
and here are the two overlapping:
![image](https://github.com/IntelRealSense/realsense-ros/assets/79418773/c92a5f85-f01c-4c0c-b487-cb220b215b9f)
Sorry for the confusion

**Comment by MartyG-RealSense** [comment]
It's no problem at all.  :)

The mesh image looks correct.  base_link corresponds to the tripod screw thread on the base of the camera and camera_link corresponds to the center-line of the left infrared sensor on the front of the camera.

On RealSense cameras with a glass front such as D435, depth is measured from the front glass and not from the sensor inside the casing.  The depth measurement is 0 at the glass front.  To find the 'ground truth' value (the depth when measured from the sensor's lens) on a D435 camera, -3.2 (mm) is added to the depth measurement.

**Comment by samibouziri** [comment]
I am not talking about the frames or the measurements. I am talking about the position of the **collision box**. The **collision box** of the camera should encompass the camera mesh which is not the case in the picture above. 
The collision box should not be centered around the camera frame (x coordinate of the origin of the collision box 0) as the camera is not positioned in the center of the camera.
In other words when visualizing both the mesh and the collision box I am expecting to see something like this :
![image](https://github.com/IntelRealSense/realsense-ros/assets/79418773/f67450aa-1080-41c8-92e1-f3a4184ed811)
and **not** something like this:
![image](https://github.com/IntelRealSense/realsense-ros/assets/79418773/6af0a453-8936-4410-8322-60b2d73fb6df)

**Comment by MartyG-RealSense** [comment]
I have not seen this particular problem in past RealSense ROS cases, unfortunately.

Are you using ROS1 or ROS2?  You linked to the ROS2 wrapper's D435 URDF.  If you are using the ROS1 wrapper then it has a D435 URDF here:

https://github.com/IntelRealSense/realsense-ros/tree/ros1-legacy/realsense2_description/urdf

**Comment by samibouziri** [comment]
I am using ROS1 but the problem should be the same for ROS2 since the URDFs are the same and the meshes are the same. 
Anyway if you want to align the collision box with the mesh like shown just replace the collision tag which is now:

```
<collision>
        <origin xyz="0 ${-d435_cam_depth_py} 0" rpy="0 0 0"/>
        <geometry>
          <box size="${d435_cam_depth} ${d435_cam_width} ${d435_cam_height}"/>
        </geometry>
</collision>
``` 

by:

```
<collision>
        <origin xyz="${-d435_cam_depth / 2 + d435_zero_depth_to_glass + d435_glass_to_front instead} ${-d435_cam_depth_py} 0" rpy="0 0 0"/>
        <geometry>
          <box size="${d435_cam_depth} ${d435_cam_width} ${d435_cam_height}"/>
        </geometry>
</collision>
```

**Comment by MartyG-RealSense** [comment]
It's great to hear that you achieved a solution.  Thanks so much for sharing it!
